Revisit camel-quarkus bootstrap (#1344)
apache/camel-quarkus: docs/modules/ROOT/pages/user-guide/bootstrap.adoc

= Bootstrap Camel Quarkus

> :page-aliases: bootstrap.adoc

Reviewer: ```suggestion
```
If bootstrap.adoc is  a new page, there is no point in adding an alias. Aliases serve for setting redirects and we use them mainly when we move a page.

> Camel Quarkus automatically configure and deploy a Camel Context bean which by default is started/stopped according to the Quarkus Application lifecycle. The configuration step happens at build time during Quarkus' augmentation phase and it is driven by the Camel Quarkus extensions which can be tuned using Camel Quarkus specific properties indentified by "camel.quarkus" as prefix. After the configuration is done, a minimal Camel Runtime is assembled and started at RUNTIME_INIT time. At that point, any additional configuration should be done using CDI, below an example bout how to configure a Camel Component:

Reviewer: ```suggestion
Camel Quarkus automatically configures and deploys a Camel Context bean which by default is started/stopped according to the Quarkus Application lifecycle. The configuration step happens at build time during Quarkus' augmentation phase and it is driven by the Camel Quarkus extensions which can be tuned using Camel Quarkus specific `camel.quarkus.*` properties. After the configuration is done, a minimal Camel Runtime is assembled and started at `RUNTIME_INIT` time. At that point, any additional configuration should be done using CDI. Below is an example showing how to configure a Camel Component:
```

Author: I haven't yet marked the `documentation (user guide)` as done,
This is just a bunch of notes and I've committed it as I will continue on another computer so I need a way to synk them :)

[source,java]
----

> @ApplicationScoped

Reviewer: Please add all or at least the most important imports.

public class Configurations {
    /**
     * Produces a {@link LogComponent} instance with a custom exchange formatter.
     */
    @Named
    LogComponent log() {
        DefaultExchangeFormatter formatter = new DefaultExchangeFormatter();
        formatter.setShowExchangePattern(false);
        formatter.setShowBodyType(false);

        LogComponent component = new LogComponent();
        component.setExchangeFormatter(formatter);

        return component;
    }
}
----

> To configure component and other aspects of Apache Camel through properties, you can add the camel-core-main extension which brings functionnalities from https://camel.apache.org/components/latest/others/main.html[Apache Camel Main] to Camel Quarkus. By adding such extension the programmatic configuration of the Log Component done above, can be done through properties:

Reviewer: ```suggestion
To configure component and other aspects of Apache Camel through properties, you can add the `camel-core-main` extension which brings functionalities from https://camel.apache.org/components/latest/others/main.html[Apache Camel Main] to Camel Quarkus. By adding such extension the programmatic configuration of the Log Component done above, can be done through properties:
```
